### PR TITLE
Add Central Package Version support

### PR DESCRIPTION
**Pull request by davkean**
Fixes: https://github.com/dotnet/project-system/issues/5886

This adds project-system support for central package versions: https://github.com/NuGet/Home/blob/zivkan-cpvm-vs-management/designs/CentrallyManagingNuGetPackageVersions-VSManagePackages.md.

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/Snapshots/RestoreBuilder.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ namespace Microsoft.VisualStudio.ProjectSystem.VS.PackageRestore
     /// </summary>
     internal static class RestoreBuilder
     {
-        public static readonly IVsTargetFrameworks2 EmptyTargetFrameworks = new TargetFrameworks(Enumerable.Empty<IVsTargetFrameworkInfo2>());
+        public static readonly IVsTargetFrameworks2 EmptyTargetFrameworks = new TargetFrameworks(Enumerable.Empty<IVsTargetFrameworkInfo3>());
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/Snapshots/RestoreBuilder.cs]: Should the field's type be changed too?

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/Snapshots/RestoreBuilder.cs]: No, they didn't version this type. It was a mistake that it was versioned from IVsTargetFrameworks -> IVsTargetFrameworks2 last time.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/CollectedPackageVersion.xaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    <DataSource HasConfigurationCondition="False"
+                ItemType="PackageVersion"
+                MSBuildTarget="CollectCentralPackageVersions"
+                Persistence="ProjectFile"
+                SourceOfDefaultValue="AfterContext"
+                SourceType="TargetResults" />
+  </Rule.DataSource>
+
+  <StringProperty Name="Version"
+                  DisplayName="Version" />
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/CollectedPackageVersion.xaml]:
```suggestion
  <StringProperty Name="Version"
                  Visible="false" />
```

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/Snapshots/TargetFrameworkInfo.cs [hunk 3]
```diff
@@ -9,16 +9,17 @@ namespace Microsoft.VisualStudio.ProjectSystem.VS.PackageRestore
...
         // If additional fields/properties are added to this class, please update RestoreHasher
-        public TargetFrameworkInfo(string targetFrameworkMoniker, IVsReferenceItems frameworkReferences, IVsReferenceItems packageDownloads, IVsReferenceItems projectReferences, IVsReferenceItems packageReferences, IVsProjectProperties properties)
+        public TargetFrameworkInfo(string targetFrameworkMoniker, IVsReferenceItems frameworkReferences, IVsReferenceItems packageDownloads, IVsReferenceItems projectReferences, IVsReferenceItems packageReferences, IVsReferenceItems centralPackageVersions, IVsProjectProperties properties)
         {
             TargetFrameworkMoniker = targetFrameworkMoniker;
             FrameworkReferences = frameworkReferences;
             PackageDownloads = packageDownloads;
             ProjectReferences = projectReferences;
             PackageReferences = packageReferences;
+            CentralPackageVersions = centralPackageVersions;
```
Comment (davidwengier) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/Snapshots/TargetFrameworkInfo.cs]: Nit: Are these "Central"? They're just PackageVersion items, right? Can a project not use PackageVersion items to control the version of transient packages, regardless of whether they're defined centrally or not?

Comment (davidwengier) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/Snapshots/TargetFrameworkInfo.cs]: Okay, I see the msbuild task has Central in the name, so i guess this is fine.. though maybe that should change  too :)

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/Snapshots/TargetFrameworkInfo.cs]: This is an interface so I have to use this name, feedback to @cristinamanum.

**Pull request merged**